### DISCUSSION TITLE
How to set custom `user-agent` header and allow redirects?

### DISCUSSION THREAD

**Discussion by higoka** [nodejs/undici#1182] [Q&A]
How can i set a custom `user-agent` header for requests?
Also is it possible to allow redirections?

I find the docs very hard for new `undici` users like me

**Comment by Eomm** [comment]
You can provide your headers to the `request` method:

```js
    const client = new Client(`http://localhost:${server.address().port}`)
    const { statusCode, body } = await client.request({
      path: '/',
      method: 'GET',
      headers: {
        'user-agent': 'foo bar'
      }
    })
```

Checkout the `maxRedirections` for redirects
